extensions/data-transfer/portability-data-transfer-microsoft/src/main/java/org/dataportabilityproject/transfer/microsoft/MicrosoftTransferExtension.java

```diff
@@ -55,6 +65,10 @@ public String getServiceId() {
       return new MicrosoftPhotosExporter(BASE_GRAPH_URL, client, mapper, jobStore);
     }
 
+    if (derivedData && transferDataType.equals(DERIVED_DATA)) {
```

Comment:
can you add a comment why the flag is here for derived data and not the other verticals?

Reply:
Sure

Reply:
sorry wasn't clear, I meant why derived needs a flag but the other data types don't?

Reply:
For now, I think so. I added a TODO about potentially adding a configuration option that allows builders/deployers to disable (or enable) transfer of certain data types. For example, "baseline" data types could be turned on by default while "experimental/optional" could be off.

It looks as if the number of config switches is quickly multiplying and it may be getting unwieldy to figure out how to specify all of them. One way around this would be to create a Gradle plugin that creates a configured build artifact. For example, someone would use the plugin in their own Gradle build to handle all of the configuration setup for the runtime. The Gradle build would have a project in it that depended on the relevant data portability modules and would have plugins applied to generate Docker images. In addition, the project would apply the "data portability plugin" and generate all of the required YAML, command line settings, etc. The plugin would provide one typed way to view and specify runtime settings. 

What do you think? That's probably something we should all discuss and think about in relation to the current config.

Thanks!